parcels/collection/collectionsoa.py

```diff
@@ -980,6 +980,14 @@ def __init__(self, pcoll, reverse=False, subset=None):
     def __iter__(self):
         return ParticleCollectionIteratorSOA(pcoll=self._pcoll_immutable, reverse=self._reverse, subset=self._subset)
 
+    def __len__(self):
+        """Implementation needed for particle-particle interaction"""
+        return len(self._subset)
+
+    def __getitem__(self, items):
+        """Implementation needed for particle-particle interaction"""
+        return ParticleAccessorSOA(self._pcoll_immutable, self._subset[items])
```

Comment:
why or where is that needed ? The separation between an `Iterable` and an `Iterator` was motivated exactly because having a `__getitem__(...)` function doesn't work well in `for item in collection:` loops. So reintroducing them rather hints to me something is a bit odd in the code. Can you tell me why re-adding this was required ?

Reply:
It allows for users to directly access the i-th value of the neighbors list. So, for example in the `test_interaction_kernel.py` file, the DummyMoveNeighbor kernel only has to loop over the neighbors once to get the closest neighbor, and apply the mutation, instead of having to loop twice. So basically it's a convenience/performance feature. It can be removed if that's not useful.

Reply:
I'll look at it again, taking this feedback into account. Thanks.

Reply:
I saw the way it works right now in the InteractionKernel dummy - basically, your neighbours are (indexable) iterables. That holds true for SoA (and also AoS), so in that case this implementation works indeed. As the code is also here in the specialized class, I think it's good as-is.